### PR TITLE
Stripping comments from gcode before forwarding to machines.

### PR DESCRIPTION
**Pull request by gsrunion**

### CODE REVIEW

**Comment by crpalmer** [comment]
The comment matching should be done in two separate steps.  The commentMatcher is used to preserve the line comment for tool changes, pauses, etc.  So it should match and save /;.*/.  After that comment has been removed, the (...) comments can then be removed.

The regex that you are using for the (...) comments is incorrect.  It should be /\([^\)]*\)/gm;.  The command

X (comment) Y (comment ( ( ( comment) Z

should be X Y Z when you're done removing comments.  Your regex would incorrectly make it X Z.

If this were my code repository, I would want to see this as 2 comments.  The first changing the comments and the second adding in the extra regex.  You could also consider adding a filtering step which removes all the (...) comments before returning line at the end of Feeder and Sender.

**Comment by gsrunion** [comment]
> The comment matching should be done in two separate steps. The commentMatcher is used to preserve the line comment for tool changes, pauses, etc. So it should match and save /;.*/. After that comment has been removed, the (...) comments can then be removed.
> 
> The regex that you are using for the (...) comments is incorrect. It should be /([^\)]*)/gm;. The command
> 
> X (comment) Y (comment ( ( ( comment) Z
> 
> should be X Y Z when you're done removing comments. Your regex would incorrectly make it X Z.
> 
> If this were my code repository, I would want to see this as 2 comments. The first changing the comments and the second adding in the extra regex. You could also consider adding a filtering step which removes all the (...) comments before returning line at the end of Feeder and Sender.

Ok. I'll look into making those changes. JS is not my strong suite, but I'll manage.

File: src/server/controllers/Grbl/GrblController.js [hunk 1]
```diff
@@ -373,7 +373,8 @@ class GrblController {
                     });
                 }
 
-                return line;
+                // remove all comments before forwarding on
+                return line.replace(/([^\)]*)/gm, '').replace(commentMatcher, '');
```
Comment (GibsT) [inline src/server/controllers/Grbl/GrblController.js:377]: Change this to return line.replace(/\([^\)]*\)/gm, '').trim();

File: src/server/controllers/Grbl/GrblController.js [hunk 2]
```diff
@@ -540,7 +541,8 @@ class GrblController {
                  * End of Rotary Logic
                  */
 
-                return line;
+                // remove all comments before forwarding on
+                return line.replace(/([^\)]*)/gm, '').replace(commentMatcher, '');
```
Comment (GibsT) [inline src/server/controllers/Grbl/GrblController.js:545]: Change this to return line.replace(/\([^\)]*\)/gm, '').trim();

File: src/server/controllers/Grblhal/GrblHalController.js [hunk 3]
```diff
@@ -336,7 +336,8 @@ class GrblHalController {
                     line = line.replace('M6', ';M6');
                 }
 
-                return line;
+                // remove all comments before forwarding on
+                return line.replace(/([^\)]*)/gm, '').replace(commentMatcher, '');
```
Comment (GibsT) [inline src/server/controllers/Grblhal/GrblHalController.js:340]: Change this to return line.replace(/\([^\)]*\)/gm, '').trim();

File: src/server/controllers/Grblhal/GrblHalController.js [hunk 4]
```diff
@@ -479,7 +480,8 @@ class GrblHalController {
                     line = line.replace(`${tool?.[0]}`, `(${tool?.[0]})`);
                 }
 
-                return line;
+                // remove all comments before forwarding on
+                return line.replace(/([^\)]*)/gm, '').replace(commentMatcher, '');
```
Comment (GibsT) [inline src/server/controllers/Grblhal/GrblHalController.js:484]: Change this to return line.replace(/\([^\)]*\)/gm, '').trim();

**Review by GibsT — CHANGES_REQUESTED** [review]
1. Regex is missing escape character for parenthesis. 
2. Using replace(commentMatcher, '') is redundant as it is already done previously.

File: src/server/controllers/Grbl/GrblController.js (hunk 1 above)
Comment (gsrunion) [reply src/server/controllers/Grbl/GrblController.js]: https://github.com/Sienci-Labs/gsender/pull/448/commits/c609723ecffe93c6ce104906b192a6f8be2ffdea

File: src/server/controllers/Grbl/GrblController.js (hunk 2 above)
Comment (gsrunion) [reply src/server/controllers/Grbl/GrblController.js]: https://github.com/Sienci-Labs/gsender/pull/448/commits/c609723ecffe93c6ce104906b192a6f8be2ffdea

File: src/server/controllers/Grblhal/GrblHalController.js (hunk 3 above)
Comment (gsrunion) [reply src/server/controllers/Grblhal/GrblHalController.js]: https://github.com/Sienci-Labs/gsender/pull/448/commits/c609723ecffe93c6ce104906b192a6f8be2ffdea

File: src/server/controllers/Grblhal/GrblHalController.js (hunk 4 above)
Comment (gsrunion) [reply src/server/controllers/Grblhal/GrblHalController.js]: https://github.com/Sienci-Labs/gsender/pull/448/commits/c609723ecffe93c6ce104906b192a6f8be2ffdea

**Pull request closed without merging**